### ISSUE TITLE
Wrongly reported syntax error prevents upload

### ISSUE THREAD

**Issue by NiklasGollenstede** [mozilla/addons-linter#1325]
### Describe the problem and steps to reproduce it:
When I try to upload the attached .zip, I get
![image](https://user-images.githubusercontent.com/10552031/26881841-ed4b4fbe-4b98-11e7-84b3-0283ace7140e.png)
in this file
https://github.com/NiklasGollenstede/web-ext-utils/blob/46a0800cbc2f042143a37a498c736c37b1eba76a/utils/event.js#L41
which means I am unable to update my extension.
But that file is perfectly fine syntax wise, it loads and does what it's supposed to do.

### What happened?
I can't upload due to that "syntax error".

### What did you expect to happen?
I can upload because there is no error, an even if there was, I don't think it should stop me from uploading (at least to the dev channel).

### Anything else we should know?
Here is the exact file I wanted to upload:
[wikipedia_peek-2.8.5b18.zip](https://github.com/mozilla/addons-linter/files/1058176/wikipedia_peek-2.8.5b18.zip)

**Comment by wagnerand** [comment]
I looked into this and it seems because of the usage of `async` as an object property. I'm not sure why though, `async` is not a reserved word in JS.

**Comment by EnTeQuAk** [comment]
I found https://github.com/eslint/espree/issues/322 - and we are using `ecmaVersion: 2017` which is equivalent to `8` which means, the linter actually supports async functions and tries to interpret them.

I'm about to say that this is something that should not be fixed but rather on the users end.

**Comment by EnTeQuAk** [comment]
Firefox supports it with version 52 too, https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function - so I'm going to close this since `async` here is a reserved keyword and should not be used.

**Comment by EnTeQuAk** [comment]
Hmm, or maybe this is a problem, theoretically it should not appear reading https://github.com/eslint/espree/pull/295

**Comment by muffinresearch** [comment]
Looks like a bug in espree related to spreading `async` as an object prop. Eslint directly shows the same problem using it directly. 

Reduced test-case:

```
/* global console */

const async = false;
const bar = 1;
const foo = { async, bar };

console.log(foo);
```

**Comment by muffinresearch** [comment]
I've filed: https://github.com/eslint/espree/issues/340

**Comment by EnTeQuAk** [comment]
And this is about to land upstream, yeay! Thanks for filing this @muffinresearch!

**Comment by EnTeQuAk** [comment]
So https://github.com/eslint/espree/commit/be85b8e97eafc502f9d3150eb5d324639d308051 landed in master, once espree get's a new release we can upgrade it and this will be fixed once that happens.

**Comment by EnTeQuAk** [comment]
This should have been landed in espree 3.5.0 and eslint 4.7.1 pulls in espree 3.5.1 so this should be fixed in the latest releases :)

**Comment by ValentinaPC** [comment]
@EnTeQuAk : checking submission for the attached extension and seeing that is working, should do it for this? Or there is anything else that should be checked here? Thanks!

**Comment by EnTeQuAk** [comment]
Yes, that should suffice.

**Comment by ValentinaPC** [comment]
Verified as fixed on AMO-dev FF55(Win 7). 
Submission of the attached extensions doesn't return any error.
Postfix screenshot:
![sub](https://user-images.githubusercontent.com/15685960/30805825-e72109f8-a1fb-11e7-8b07-e047a649b9fb.png)
